### PR TITLE
JS3 week3 

### PR DESCRIPTION
**Pull request by mehmetekinci**

### CODE REVIEW

File: homework/index.js [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
-      if (xhr.status >= 200 && xhr.status <= 299) {
-        cb(null, xhr.response);
-      } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
-      }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
+  async function fetchJSON(url) {
+    const response = await fetch(url);
```
Comment (remarcmij) [inline homework/index.js]: - [x] Use async/await
- [x] Use fetch

File: homework/index.js [hunk 2]
```diff
@@ -1,19 +1,28 @@
...
+    if (!response.ok) {
+      throw new Error(
+        `Network error: ${response.status} - ${response.statusText}`,
+      );
+    }
+    return response.json();
+    // return new Promise((resolve, reject) => {
```
Comment (remarcmij) [inline homework/index.js]: "Dead code" should never be left in a PR. In real life, a PR is a request to pull your changes into the production master branch. Nobody is interested in what you tried and abandoned. You can always look at previous versions of your code through your Git commits.

File: homework/index.js [hunk 3]
```diff
@@ -29,25 +38,145 @@
...
+      });
+      createAndAppend('span', li, {
+        text: contributor.contributions,
+        class: 'span-elm',
+      });
     });
   }
 
+  async function renderContributors(selectedRepo, contributionsUl, root) {
+    try {
```
Comment (remarcmij) [inline homework/index.js]: - [x] async/await
- [x] try/catch

File: homework/index.js [hunk 4]
```diff
@@ -29,25 +38,145 @@
...
+      const contributors = await fetchJSON(selectedRepo.contributors_url);
+      contributionsUl.innerHTML = '';
+      renderContributorDetails(contributors, contributionsUl);
+    } catch (err) {
+      createAndAppend('div', root, {
+        text: err.message,
+        class: 'alert-error',
+      });
+    }
+    // fetchJSON(selectedRepo.contributors_url)
```
Comment (remarcmij) [inline homework/index.js]: Dead code

File: homework/index.js [hunk 5]
```diff
@@ -29,25 +38,145 @@
...
+    });
+    const contributionsSection = createAndAppend('section', mainElm, {
+      class: 'contributions-section',
+    });
+    createAndAppend('h5', contributionsSection, {
+      text: 'Contributions',
+      class: 'contributions-title',
+    });
+    const contributionsUl = createAndAppend('ul', contributionsSection);
+    try {
```
Comment (remarcmij) [inline homework/index.js]: - [x] async/await
- [x] try/catch

File: homework/index.js [hunk 6]
```diff
@@ -29,25 +38,145 @@
...
+      );
+      renderPage(repos[0], reposSection, contributionsUl);
+    } catch (err) {
+      createAndAppend('div', root, {
+        text: err.message,
+        class: 'alert-error',
+      });
+    }
+
+    // fetchJSON(url).then(repos => {
```
Comment (remarcmij) [inline homework/index.js]: Dead code

File: homework/index.js (hunk 2 above)
Comment (mehmetekinci) [reply homework/index.js]: I am sorry @remarcmij . Next time I take account. Thank you.

**Comment by mehmetekinci** [comment]
hi @remarcmij . I completed homework.

**Comment by mehmetekinci** [comment]
> Hi @mehmetekinci, looks good! Some methods could be made **static**, but I'll explain that in the recap session tomorrow. I'm happy to approve this PR as-is.

@remarcmij Thank you for all what you teach us.

**Pull request closed without merging**